lib/vdsm/storage/blockVolume.py

```diff
@@ -164,7 +164,16 @@ def getParent(self):
         """
         Return parent volume UUID
         """
-        return self.getParentTag()
+        parent_tag = self.getParentTag()
+        parent_meta = self.getParentMeta()
+        if parent_tag != parent_meta:
+            self.log.warning(
+                "volUUID=%s parent in tag (%s) and metadata (%s) differ",
+                self.volUUID, parent_tag, parent_meta)
+            lvm.reload_lvs(self.sdUUID)
```

Comment:
Using the current API:

    lvm.invalidateVG(self.sdUUID)

will have the same affect - when we call getParentMeta we will get a fresh lv with the current tags.